### PR TITLE
Remove not used methods TWAnySignerDecode and TWAnySignerEncode

### LINKED ISSUE #1359: [AnySigner] remove not used encode method
`TWAnySignerDecode` and `TWAnySignerEncode`, not used in the app at all

### PR DESCRIPTION
**Pull request by catenocrypt**
## Description

Remove not used methods TWAnySignerDecode and TWAnySignerEncode methods.  Fixes #1359 .
Note: coins.md change is not related, but it is out of sync for a while now, fixing that.

## Checklist

- [ ] Prefix PR title with `[WIP]` if necessary.
- [ ] Add tests to cover changes as needed.
- [ ] Update documentation as needed.
- [ ] I have read the [guidelines](https://developer.trustwallet.com/wallet-core/newblockchain#integration-criteria) for adding a new blockchain
- [ ] If there is a related Issue, mention it in the description (e.g. Fixes #<issue_number> ).

### CODE REVIEW

**Comment by catenocrypt** [comment]
Removed most what was added with #962, coverage should be non-decreasing

**Review by vikmeup — APPROVED** [review]
Would this break Trust SDK?

File: docs/coins.md [hunk 1]
```diff
@@ -52,7 +52,7 @@ This list is generated from [./coins.json](../coins.json)
 | 820     | Callisto         | CLO    | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/callisto/info/logo.png" width="32" />     | <https://callisto.network>    |
 | 888     | NEO              | NEO    | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/neo/info/logo.png" width="32" />          | <https://neo.org>             |
 | 889     | TomoChain        | TOMO   | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/tomochain/info/logo.png" width="32" />    | <https://tomochain.com>       |
-| 966     | Polygon          | MATIC  | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/matic/info/logo.png" width="32" />        | <https://polygon.technology>  |
+| 966     | Polygon          | MATIC  | <img src="https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/polygon/info/logo.png" width="32" />      | <https://polygon.technology>  |
```
Comment (hewigovens) [inline docs/coins.md:55]: @catenocrypt let's change asset repo folder to polygon as well?

File: src/Ethereum/RLP.cpp [hunk 2]
```diff
@@ -140,137 +134,3 @@ Data RLP::putint(uint64_t i) noexcept {
...
-        if (listItem.remainder.size() == 0) {
-            break;
-        } else {
-            remainder = listItem.remainder;
-        }
-    }
-    return item;
-}
-
-static uint64_t decodeLength(const Data& data) {
```
Comment (hewigovens) [inline src/Ethereum/RLP.cpp:178]: I'd like to keep RLP basic type decoding (tx decoding can be removed), what do you think?

File: docs/coins.md (hunk 1 above)
Comment (catenocrypt) [reply docs/coins.md:55]: Good point, just did it

File: src/Ethereum/RLP.cpp (hunk 2 above)
Comment (catenocrypt) [reply src/Ethereum/RLP.cpp:178]: I've put back the decodeList, decodeLenght, and decode(rest) methods.

**Comment by catenocrypt** [comment]
> Would this break Trust SDK?

I think not

**Review by hewigovens — APPROVED** [review]

**Pull request merged**